cmd/cstor-volume-grpc/app/command/commands.go

```diff
@@ -0,0 +1,280 @@
+/*
+Copyright 2018 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/cmd/cstor-volume-grpc/api"
+	"github.com/openebs/maya/pkg/util"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+)
+
+var (
+	cmdName = "cstor-volume-grpc"
+	usage   = fmt.Sprintf("%s", cmdName)
+)
+
+// CmdSnaphotOptions holds the options for snapshot
+// create command
+type CmdSnaphotOptions struct {
+	volName  string
+	snapName string
+}
+
+// Validate validates the flag values
+func (c *CmdSnaphotOptions) Validate(cmd *cobra.Command) error {
+	if c.volName == "" {
+		return errors.New("--volname is missing. Please specify an unique name")
+	}
+	if c.snapName == "" {
+		return errors.New("--snapname is missing. Please specify an unique name")
+	}
+
+	return nil
+}
+
+//CreateSnapshot creates snapshots
+func CreateSnapshot(volName, snapName string) (*api.VolumeCommand, error) {
+	var conn *grpc.ClientConn
+	conn, err := grpc.Dial(fmt.Sprintf(":%d", api.VolumeGrpcListenPort), grpc.WithInsecure())
+	if err != nil {
+		log.Fatalf("did not connect: %s", err)
+	}
+	defer conn.Close()
+
+	c := api.NewRunCommandClient(conn)
+	response, err := c.RunVolumeCommand(context.Background(),
+		&api.VolumeCommand{
+			Command:  api.CmdSnapCreate,
+			Volume:   volName,
+			Snapname: snapName,
+			Status:   "requesting",
```

Comment:
What other states are there? This seems arbitrary why are these states not defined in the protocol?

Reply:
we do not need to send anything in status field for requests. only the response has the status field set to the status of the command whether success or failure. so, i am leaving this field as empty now.

Reply:
Well, you are kind of adding something random in this status field which should be empty -- that makes no sense at all. 

If your REQ and REP message are different, then you should make different messages. If you are -- for whatever reason, not interested in doing that, then at least give it a default value from within the proto spec.

Reply:
ok. will change the REQ and RESP as different data structures in proto spec and use the same in code to send the request and response